liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -752,7 +752,7 @@ protected void doExecute() throws Exception {
         // which is where the server.xml is located if a specific serverXmlFile
         // configuration parameter is not specified.
         try {
-            util.watchFiles(pom, outputDirectory, testOutputDirectory, executor, artifactPaths, serverXmlFile);
+            util.watchFiles(pom, outputDirectory, testOutputDirectory, executor, artifactPaths, serverXmlFile, bootstrapPropertiesFile, jvmOptionsFile, serverEnvFile);
```

Comment:
If jvmOptions, bootstrapProperties or serverEnv properties are specified in the pom.xml configuration, then those take precedence and any File specified for these three parameters you are passing in should be ignored. Does the code in ci.common take that into account?

Reply:
Not entirely at the moment. The values in the files never take precedence, however, the code still triggers a server restart if the file is modified, but the properties are specified in the pom.xml. @ericglau and I had discussed this, and thought it wasn't a major issue. What do you think?

Reply:
And there is more than one way those properties can be specified. First is through specific configuration parameters, and second is through maven properties with a specific naming format (liberty.bootstrap.xyz, liberty.jvm.abc, liberty.env.lmn, etc). The maven properties can be specified in pom.xml or through the command line invocation with -D. See documentation here: (https://github.com/OpenLiberty/ci.maven/blob/master/docs/common-server-parameters.md#common-server-parameters)

Reply:
@cherylking In the scenarios you mentioned, if the user specifies configuration parameters that should take precedence but then updates a File that was specified but should be ignored, then dev mode will restart the server (which is unnecessary), but the actual values that are used will not be updated.  We will open a separate issue to avoid dev mode restarting the server unnecessarily in this case.